### PR TITLE
feat: date generator functions

### PR DESCRIPTION
**Pull request by mathnogueira**
This PR adds support for two new generator functions: `date()` and `dateTime()`.

## Motivation
This was asked by a user in our [discord channel](https://discordapp.com/channels/884464549347074049/963470167327772703/1169196012120834130)

## About the functions
```go
// returns the date in the time.DateOnly format (YYYY-MM-DD)
// if format is provided, it will be formatted based on it.
date(format optional<string>)

// returns the date and time in the time.RFC3339 format (2006-01-02T15:04:05Z07:00)
// if format is provided, it will be formatted based on it.
dateTime(format optional<string>)
```

## Examples

```
date() : 2023-11-01
date("MM-DD-YYYY") : 11-01-2023
date("DD/MM/YYYY") : 01/11/2023
dateTime() : 2023-11-01T12:38:12Z03:00
dateTime("hh:mm:ss") : 12:38:12
```

## Checklist

- [x] tested locally
- [x] added new dependencies
- [x] updated the docs
- [x] added a test

## Loom video

Add your loom video here if your work can be visualized

### CODE REVIEW

File: docs/docs/cli/creating-tests.mdx [hunk 1]
```diff
@@ -169,17 +169,23 @@ Generator functions can be invoked as part of expressions. Therefore, you only n
...
+| `phone()`             | Generates a random phone number.                                                              |
+| `creditCard()`        | Generates a random credit card number (from 12 to 19 digits).                                 |
+| `creditCardCvv()`     | Generates a random credit card cvv (3 digits).                                                |
+| `creditCardExpDate()` | Generates a random credit card expiration date (mm/yy).                                       |
+| `randomInt(min, max)` | Generates a random integer contained in the closed interval defined by [`min`, `max`].        |
+| `date(format?)`       | Get the current date and formats it. Default is `YYYY-MM-DD` but you can specify other formats|
+| `dateTime(format?)`   | Get the current datetime and formats it. Default is RFC3339 but you can specify other formats |
+
+:::tip
+[Continue reading about date and datetime formats, here.](https://www.w3.org/TR/NOTE-datetime)
```
Comment (jfermi) [inline docs/docs/cli/creating-tests.mdx:187]:
```suggestion
[Continue reading about date and datetime formats here.](https://www.w3.org/TR/NOTE-datetime)
```

File: docs/docs/cli/creating-tests.mdx [hunk 2]
```diff
@@ -169,17 +169,23 @@ Generator functions can be invoked as part of expressions. Therefore, you only n
...
+| `lastName()`          | Generates a random English last name.                                                         |
+| `fullName()`          | Generates a random English first and last name.                                               |
+| `email()`             | Generates a random email address.                                                             |
+| `phone()`             | Generates a random phone number.                                                              |
+| `creditCard()`        | Generates a random credit card number (from 12 to 19 digits).                                 |
+| `creditCardCvv()`     | Generates a random credit card cvv (3 digits).                                                |
+| `creditCardExpDate()` | Generates a random credit card expiration date (mm/yy).                                       |
+| `randomInt(min, max)` | Generates a random integer contained in the closed interval defined by [`min`, `max`].        |
+| `date(format?)`       | Get the current date and formats it. Default is `YYYY-MM-DD` but you can specify other formats|
```
Comment (jfermi) [inline docs/docs/cli/creating-tests.mdx:183]:
```suggestion
| `date(format?)`       | Get the current date and formats it. Default is `YYYY-MM-DD` but you can specify other formats.|
```

File: docs/docs/cli/creating-tests.mdx [hunk 3]
```diff
@@ -169,17 +169,23 @@ Generator functions can be invoked as part of expressions. Therefore, you only n
...
+| `dateTime(format?)`   | Get the current datetime and formats it. Default is RFC3339 but you can specify other formats |
```
Comment (jfermi) [inline docs/docs/cli/creating-tests.mdx:184]:
```suggestion
| `dateTime(format?)`   | Get the current datetime and formats it. Default is RFC3339 but you can specify other formats.|
```

File: docs/docs/cli/creating-tests.mdx [hunk 4]
```diff
@@ -169,17 +169,23 @@ Generator functions can be invoked as part of expressions. Therefore, you only n
...
+
+:::tip
+[Continue reading about date and datetime formats, here.](https://www.w3.org/TR/NOTE-datetime)
+:::
 
 :::tip
 [Continue reading about Test Specs, here.](/cli/creating-test-specifications)
```
Comment (jfermi) [inline docs/docs/cli/creating-tests.mdx:191]:
```suggestion
[Continue reading about Test Specs here.](/cli/creating-test-specifications)
```

**Review by jfermi — COMMENTED** [review]
Some editing suggestions.

**Review by adnanrahic — APPROVED** [review]

**Review by danielbdias — APPROVED** [review]

**Pull request merged**